src/pal/src/exception/seh-unwind.cpp

```diff
@@ -251,6 +254,11 @@ BOOL PAL_VirtualUnwind(CONTEXT *context, KNONVOLATILE_CONTEXT_POINTERS *contextP
         return FALSE;
     }
 #endif
+
+#if !UNWIND_CONTEXT_IS_UCONTEXT_T
+    UpdateUnwindContextWithWinContext(context, &unwContext);
+#endif
+
```

Comment:
More specifically, ARM64, AMD64, and I386 are UNWIND_CONTEXT_IS_UCONTEXT_T = 1. Only ARM has 0.
(header files of libunwind have explanations on this)

You do not need to change the macros because of this, but you can make the code more readable based on this fact.
I.e., the line 245:263 (or 246:271 after this patch) may be updated to have a single #if-#else-#endif structure while removing duplicated operations.

Anyway, you are right about the last replies, this patch does have the correctness. Please refactor/consolidate the logic here, which is going to be even more distorted after this.